datapackage_pipelines/lib/dump/dumper_base.py

```diff
@@ -194,23 +195,24 @@ def copy_non_tabular_resources(self, datapackage):
         for resource in datapackage['resources']:
             if not streaming(resource):
                 url = resource[PROP_STREAMED_FROM]
-                delete = False
+                tmp = tempfile.NamedTemporaryFile(delete=False)
                 if is_a_url(url):
-                    tmp = tempfile.NamedTemporaryFile(delete=False)
                     stream = requests.get(url, stream=True).raw
                     stream.read = functools.partial(stream.read, decode_content=True)
                     shutil.copyfileobj(stream, tmp)
-                    filesize = tmp.tell()
-                    tmp.close()
-                    url = tmp.name
-                    delete = True
                 else:
-                    filesize = os.stat(url).st_size
+                    shutil.copyfile(url, tmp.name)
```

Comment:
Why copy a local file to a temporary location?

Reply:
@akariv thought would be better than loading it into memory (as we have to open to get hash)

Reply:
Hm... realised that anyway have to do that :)